src/main/java/org/ibp/api/rest/germplasmlist/GermplasmListResourceGroup.java

```diff
@@ -209,4 +228,117 @@ public ResponseEntity deleteGermplasmListFolder(
+			value = "Sorting criteria in the format: property,asc|desc. ")
+	})
+	@ResponseBody
+	public ResponseEntity<List<GermplasmListSearchResponse>> searchGermplasmLists(
+		@PathVariable final String cropName,
+		@ApiParam("The program UUID") @RequestParam(required = false) final String programUUID,
+		@RequestBody final GermplasmListSearchRequest request,
+		@ApiIgnore @PageableDefault(page = PagedResult.DEFAULT_PAGE_NUMBER, size = PagedResult.DEFAULT_PAGE_SIZE) final Pageable pageable) {
+		return new PaginatedSearch().getPagedResult(() -> this.germplasmListService.countSearchGermplasmList(request, programUUID),
+			() -> this.germplasmListService.searchGermplasmList(request, pageable, programUUID),
+			pageable);
+	}
+
+	@ApiOperation(value = "Returns a germplasm list data by a given germplasm list id")
+	@PreAuthorize("hasAnyAuthority('ADMIN', 'LISTS', 'GERMPLASM_LISTS', 'MANAGE_GERMPLASM_LISTS', 'SEARCH_GERMPLASM_LISTS')")
+	@RequestMapping(value = "/crops/{cropName}/germplasm-lists/{listId}/data/search", method = RequestMethod.POST)
+	@ApiImplicitParams({
+		@ApiImplicitParam(name = "page", dataType = "integer", paramType = "query",
+			value = "page number. Start at " + PagedResult.DEFAULT_PAGE_NUMBER),
+		@ApiImplicitParam(name = "size", dataType = "integer", paramType = "query",
+			value = "Number of records per page."),
+		@ApiImplicitParam(name = "sort", allowMultiple = false, dataType = "string", paramType = "query",
+			value = "Sorting criteria in the format: property,asc|desc. ")
+	})
+	@ResponseBody
+	public ResponseEntity<List<GermplasmListDataSearchResponse>> searchGermplasmListData(
+		@PathVariable final String cropName,
+		@PathVariable final Integer listId,
+		@ApiParam("The program UUID") @RequestParam(required = false) final String programUUID,
+		@RequestBody final GermplasmListDataSearchRequest request,
+		@ApiIgnore @PageableDefault(page = PagedResult.DEFAULT_PAGE_NUMBER, size = PagedResult.DEFAULT_PAGE_SIZE) final Pageable pageable) {
+		return new PaginatedSearch().getPagedResult(() -> this.germplasmListDataService.countSearchGermplasmListData(listId, request),
+			() -> this.germplasmListDataService.searchGermplasmListData(listId, request, pageable),
+			pageable);
+	}
+
+	@ApiOperation(value = "Returns a list by a given list id")
+	@RequestMapping(value = "/crops/{cropName}/germplasm-lists/{listId}", method = RequestMethod.GET)
+	@ResponseBody
+	public ResponseEntity<GermplasmListDto> getGermplasmListById(@PathVariable final String cropName,
+		@PathVariable final Integer listId,
+		@RequestParam(required = false) final String programUUID) {
+		return new ResponseEntity<>(this.germplasmListService.getGermplasmListById(listId), HttpStatus.OK);
+	}
+
+	//TODO: check permission
+	@ApiIgnore
+	@ApiOperation(value = "Locks or unlocks the list depending the current status of it")
+	@RequestMapping(value = "/crops/{cropName}/germplasm-lists/{listId}/toggle-status", method = RequestMethod.POST)
```

Comment:
add permissions, I will say that we add the same we are using for GET now until Mariano clarifies if we want a specific lock/unlock permission

Reply:
Done